packages/core/src/connectors/injected.ts

```diff
@@ -75,27 +75,26 @@ export class InjectedConnector extends Connector {
   }
 
   async chainId(): Promise<bigint> {
-    /*
     this.ensureWallet();
 
     if (!this._wallet) {
       throw new ConnectorNotConnectedError();
     }
 
-    const chainIdHex = await this._wallet.provider.getChainId();
+    const chainIdHex = await this._wallet.request({
+      type: "wallet_requestChainId",
+    });
```

Comment:
Notice that if the wallet is not authorized, this method will throw an exception. Wrap it into a `try catch` nd throw `ConnectorNotConnectedError`